### ISSUE TITLE
verdaccio looks for a plugin named minio-storage, while the NPM package is minio

### ISSUE THREAD

**Issue by lando-cal** [barolab/verdaccio-minio#2]
https://github.com/barolab/verdaccio-minio/blob/fa3df905e5813bb5b6f31110ae8ca1588868ae42/src/index.ts#L37

This causes Verdaccio to fail if config.yaml looks like this:
```yaml
store:
   minio-storage:
```

Conversely, the plugin fails if config.yaml looks like this:
```yaml
store:
   minio:
```

**Comment by barolab** [comment]
What do you mean by Verdaccio is failing ? Do you have any log ? Did you try the [example configuration](https://github.com/barolab/verdaccio-minio/blob/master/example/config.yaml#L16) ?

**Comment by lando-cal** [comment]
Verdaccio looks in node_modules for a package named `minio-storage` when referenced that way. This may be a new change, I only started using Verdaccio a couple of days ago, so I'm not familiar with the history. Here is a Dockerfile that demonstrates the issue.

```dockerfile
FROM verdaccio/verdaccio:4.0

USER root

RUN yarn add  verdaccio-ldap   \
              verdaccio-minio  \
              verdaccio-gitlab

ADD ./config.yaml /verdaccio/conf/config.yaml
#COPY ./minio-patch.js /opt/verdaccio/node_modules/verdaccio-minio/lib/index.js

USER verdaccio
```

As you can see in the commented line there, I've been overwriting the file in the mean time.
The only change in the patch file is in the line referenced above (or the compiled version of it, anyhow) which is now as follows:

```js
this.config = _objectSpread({}, config.store['minio']);
```
This works as expected, as long as I use the modified yaml:

```yaml
store:
   minio:
```

**Comment by barolab** [comment]
This should be fixed in the upcoming 0.1.6. But I don't think the plugin will actually work with verdaccio 4.0 because the API as changed. I'm testing with the 4.2.2, which sould be supported by both gitlab & ldap. 

Can you give me feedback on the 4.0 support ?

**Comment by lando-cal** [comment]
It worked for stored packages, but didn't seem to forward requests for packages that it couldn't find on Minio. It threw an error about not seeing them in package_name/package.json.

Local push and pull seemed to work fine though.

⁣Get BlueMail for Android ​

**Comment by barolab** [comment]
Yes, it's the behavior I have on my machine using Verdaccio 4.0. I'll try to see if I can add support, but that'll require a bit of work.

**Comment by barolab** [comment]
I think I found where the issue is. I need to do more tests in order to check the fix doesn't break the plugin for Verdaccio >4.2
I'll release a 0.2.1 in the coming days so you can test the fix.

**Comment by barolab** [comment]
Hi @lando-cal, I just released a [0.2.1](https://www.npmjs.com/package/verdaccio-minio) of the plugin, with a fix for the pacakge proxy in Verdaccio 4.0. Can you give it a try and tell me if it works for you ?

**Comment by barolab** [comment]
I’m closing this pull request for inactivity, @lando-cal feel free to re-open if you encounter another proxy problem.
